shell/platform/darwin/macos/framework/Source/FlutterCompositor.h

```diff
@@ -52,9 +53,30 @@ class FlutterCompositor {
 
  protected:
   __weak const FlutterViewController* view_controller_;
-
   PresentCallback present_callback_;
 
+  // frame_started_ keeps track of if a layer has been
+  // created for the frame.
+  bool frame_started_ = false;
+
+  // Count for how many CALayers have been created for a frame.
+  // Resets when a frame is finished.
+  // ca_layer_count_ is also used as a layerId.
+  size_t ca_layer_count_ = 0;
```

Comment:
will this always be equal to `ca_layer_map_.size()`?

Reply:
This is just refactored existing code that I moved from FlutterGLCompositor into FlutterCompositor. A closer look at it suggests that ca_layer_count_ is always going to be the size of the map; that being said, it's concerning to me that this map expands in size infinitely and is never reduced in size. I think there's a separate bug here where we need to figure out when to purge old items from the map.